### PR TITLE
Add extends prettier rules

### PR DESCRIPTION
**Pull request by infinityfarm**
Turns off all rules that are unnecessary or might conflict with prettier.

### CODE REVIEW

File: configs/react.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 module.exports = {
 	"extends": [
+		"prettier",
+		"prettier/react",
```
Comment (jrdrg) [inline configs/react.js]: Do you need `prettier/flowtype` here as well?

File: package.json [hunk 2]
```diff
@@ -26,5 +26,8 @@
   },
   "bin": {
     "dibslint": "./index.js"
+  },
+  "devDependencies": {
+    "eslint-config-prettier": "^2.6.0"
```
Comment (jrdrg) [inline package.json]: Shouldn't this be a regular dependency, not a dev dep?

**Comment by infinityfarm** [comment]
Running the `eslint-config-prettier-check` tool against our configs in `eslint-config-1stdibs` should let us know which rules extending prettier in dibslint will end up overwriting. This was the output:

## jasmine.js
```bash
$ eslint -c ./jasmine.js --print-config ./jasmine.js | eslint-config-prettier-check
No rules that are unnecessary or conflict with Prettier were found.
✨  Done in 0.48s.
```

## index.js
```bash
$ eslint -c ./index.js --print-config ./index.js | eslint-config-prettier-check
The following rules are unnecessary or might conflict with Prettier:

- array-bracket-spacing
- comma-dangle
- comma-spacing
- comma-style
- eol-last
- indent
- key-spacing
- keyword-spacing
- no-mixed-spaces-and-tabs
- no-multi-spaces
- no-spaced-func
- no-trailing-spaces
- semi
- space-before-blocks
- space-before-function-paren
- space-in-parens
- space-infix-ops
- space-unary-ops
- wrap-iife
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

## react.js
```bash
$ eslint -c ./react.js --print-config ./react.js | eslint-config-prettier-check
The following rules are unnecessary or might conflict with Prettier:

- react/jsx-closing-bracket-location
- react/jsx-indent-props
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

## es6.js
```bash
$ eslint -c ./es6.js --print-config ./es6.js | eslint-config-prettier-check
The following rules are unnecessary or might conflict with Prettier:

- flowtype/space-after-type-colon
- flowtype/union-intersection-spacing
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

## node.js
```bash
$ eslint -c ./node.js --print-config ./node.js | eslint-config-prettier-check
No rules that are unnecessary or conflict with Prettier were found.
✨  Done in 0.44s.
```

## browser.js
```bash
$ eslint -c ./browser.js --print-config ./browser.js | eslint-config-prettier-check
No rules that are unnecessary or conflict with Prettier were found.
✨  Done in 0.45s.
```

**Comment by hellatan** [comment]
what will conflict with these:
```
- array-bracket-spacing
- comma-dangle
- comma-spacing
- comma-style
```

**Comment by infinityfarm** [comment]
@hellatan comma-dangle is configurable within prettier, while the other are things that I think prettier has unconfigurable opinions about

**Comment by infinityfarm** [comment]
@hellatan any objection to me merging this and publishing with a major version bump? want to start making things _prettier_

**Comment by hellatan** [comment]
@infinityfarm is there any easy way to set up a test file to run these configs against so we can see what kind of formatting to expect?

**Comment by infinityfarm** [comment]
@hellatan have any remaining reservations about this considering that it's just shutting off lint rules that might conflict with prettier defaults and/or settings? Considering that, people should probably only bump dibslint when they add prettier, or else they could submit unlinted code.. Anyway, I'm preparing a PR prettifying abf and this is an integral part in order for dibslint to pass ;)

**Pull request merged**